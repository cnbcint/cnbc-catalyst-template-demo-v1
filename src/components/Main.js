import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import { getMargin, breakpoints } from "../helpers/Layout";
import { ModalContext } from "../components/Modal";

export const LayoutContext = React.createContext({
  maxWidth: "500px",
  colors: null,
});

const Container = styled.div`
  display: block;
  width: 100%;
  background-color: ${(props) => props.colors.background};
  > * {
    font-size: 1.125rem;
    line-height: 140%;
    color: ${(props) => props.colors.copy};
  }
  hr {
    // border-color: ${(props) => props.primaryColor};
    border-color: ${(props) => props.colors.copy};
    border-style: solid;
    display: inline-block;
    width: 100%;
    // margin-top: ${getMargin(1)};
    margin-bottom: ${getMargin(1)};
    @media (min-width: ${breakpoints.mobile}) {
      // margin-top: ${getMargin(2)};
      margin-bottom: ${getMargin(2)};
    }
  }
  img {
    width: 100%;
    display: block;
    margin-top: ${getMargin(2)};
    margin-bottom: ${getMargin(2)};
  }
  p {
    margin: 1rem auto !important;
  }
  ul {
    list-style: disc !important;
    margin-top: ${getMargin(1)} !important;
    margin-bottom: ${getMargin(1)} !important;
    li {
      list-style: disc !important;
      margin: ${getMargin(1)} !important;
      margin-left: ${getMargin(2)} !important;
    }
  }
`;
export default function Main(props) {
  return (
    <LayoutContext.Provider
      value={{ maxWidth: props.maxWidth, colors: props.colors }}
    >
      <ModalContext.Provider value={{ modals: {} }}>
        <Container
          colors={props.colors}
          primaryColor={props.colors.primary}
          className={props.className}
          style={props.style}
        >
          {props.children}
        </Container>
      </ModalContext.Provider>
    </LayoutContext.Provider>
  );
}
Main.defaultProps = {
  maxWidth: "900px",
  colors: { primary: "black", secondary: "gray", tertiary: "red" },
};
Main.propTypes = {
  className: PropTypes.string,
  children: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.array,
    PropTypes.element,
  ]).isRequired,
  maxWidth: PropTypes.string,
  colors: PropTypes.shape({
    primary: PropTypes.string,
    secondary: PropTypes.string,
    tertiary: PropTypes.string,
    background: PropTypes.string,
    copy: PropTypes.string,
  }),
  style: PropTypes.object,
};
