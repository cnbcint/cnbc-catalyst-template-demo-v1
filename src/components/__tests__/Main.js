import React from "react";
import { render } from "@testing-library/react";
import { css } from "styled-components";
import Main from "../Main";
import "jest-styled-components";

describe("Main", () => {
  it("Renders", async () => {
    const { container } = await render(<Main>Content</Main>);
    expect(container).toMatchSnapshot();
  });

  it("Children is rendered", async () => {
    const { container } = await render(<Main>Test Content</Main>);
    expect(container.firstChild.textContent).toMatch("Test Content");
  });

  it("Class name is added ", async () => {
    const { container } = await render(<Main className="test">Content</Main>);
    expect(container.firstChild.className).toContain("test");
  });

  it("Styles are added ", async () => {
    const { container } = await render(
      <Main style={{ backgroundColor: "red" }}>Content</Main>
    );
    expect(container.firstChild.style["background-color"]).toMatch("red");
  });

  it("CSS styles check", async () => {
    const { container } = await render(
      <Main colors={{ copy: "red", background: "gray" }}>
        <p id="testp">Content</p>
      </Main>
    );
    expect(container.firstChild).toHaveStyleRule("width", "100%");
    expect(container.firstChild).toHaveStyleRule("background-color", "gray");
    expect(container.firstChild).toHaveStyleRule("color", "red", {
      modifier: css`> *`,
    });
  });
});
