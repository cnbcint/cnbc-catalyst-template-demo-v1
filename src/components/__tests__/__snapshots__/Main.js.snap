// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main Renders 1`] = `
.c0 {
  display: block;
  width: 100%;
}

.c0 > * {
  font-size: 1.125rem;
  line-height: 140%;
}

.c0 hr {
  border-style: solid;
  display: inline-block;
  width: 100%;
  margin-bottom: 1rem;
}

.c0 img {
  width: 100%;
  display: block;
  margin-top: 2rem;
  margin-bottom: 2rem;
}

.c0 p {
  margin: 1rem auto !important;
}

.c0 ul {
  list-style: disc !important;
  margin-top: 1rem !important;
  margin-bottom: 1rem !important;
}

.c0 ul li {
  list-style: disc !important;
  margin: 1rem !important;
  margin-left: 2rem !important;
}

@media (min-width:768px) {
  .c0 hr {
    margin-bottom: 2rem;
  }
}

<div>
  <div
    class="c0"
  >
    Content
  </div>
</div>
`;
